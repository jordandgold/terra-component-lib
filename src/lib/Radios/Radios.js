import React, { Component } from "react";
import "./Radios.scss";
import PropTypes from "prop-types";

class Radios extends Component {
  generateRadios = () => {
    const { radios, collection } = this.props;

    return radios.map((radio, index) => {
      return (
        <div key={`radio-${index}`}>
          <label
            htmlFor={`${collection}Choice${index}`}
            className={`ter-radio__label radio-${index}`}
          >
            {radio.name}
          </label>
          <input
            className={`radio-button-${index}`}
            type="radio"
            name={collection}
            onChange={() => this.props.selectRadio(radio.name)}
            checked={this.props.selected === radio.name}
            value={radio.name}
            id={`${collection}Choice${index}`}
          />
        </div>
      );
    });
  };

  render() {
    return <div className="ter-radio-group">{this.generateRadios()}</div>;
  }
}

export default Radios;

Radios.propTypes = {
  radios: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string.isRequired
    })
  ),
  collection: PropTypes.string.isRequired,
  selected: PropTypes.string.isRequired,
  selectRadio: PropTypes.func.isRequired
};
