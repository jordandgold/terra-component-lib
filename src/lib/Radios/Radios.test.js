import React from "react";
import { shallow } from "enzyme";
import Radios from "./Radios";

describe("Radios", () => {
  let wrapper;
  let mockSelectRadio;

  beforeEach(() => {
    mockSelectRadio = jest.fn();
    const mockSelected = "test one";
    const mockRadios = ["test one", "test two", "test three"];
    const mockCollection = "test-radios";
    wrapper = shallow(
      <Radios
        radios={mockRadios}
        selectRadio={mockSelectRadio}
        selected={mockSelected}
        collection={mockCollection}
        name="test name"
      />
    );
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe("generateRadios", () => {
    it("should return some JSX", () => {
      const expected = [
        '<div><label for="test-radiosChoice0" class="ter-radio__label radio-0">test one</label><input type="radio" class="radio-button-0" name="test-radios" checked="" value="test one" id="test-radiosChoice0"/></div>',
        '<div><label for="test-radiosChoice1" class="ter-radio__label radio-1">test two</label><input type="radio" class="radio-button-1" name="test-radios" value="test two" id="test-radiosChoice1"/></div>',
        '<div><label for="test-radiosChoice2" class="ter-radio__label radio-2">test three</label><input type="radio" class="radio-button-2" name="test-radios" value="test three" id="test-radiosChoice2"/></div>'
      ];
      const radioWrappers = wrapper
        .instance()
        .generateRadios()
        .map(radio => {
          return shallow(radio).html();
        });

      expect(radioWrappers).toEqual(expected);
    });

    it("should call selectRadio on change", () => {
      const mockRadio = { name: "test one" };
      wrapper.find(".radio-button-0").simulate("change", mockRadio);

      expect(mockSelectRadio).toHaveBeenCalledWith(mockRadio.name, "test name");
    });
  });
});
