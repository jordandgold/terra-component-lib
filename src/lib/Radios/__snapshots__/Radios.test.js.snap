// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Radios should match the snapshot 1`] = `
<div
  className="ter-radio-group"
>
  <div
    key="radio-0"
  >
    <label
      className="ter-radio__label radio-0"
      htmlFor="test-radiosChoice0"
    >
      test one
    </label>
    <input
      checked={true}
      className="radio-button-0"
      id="test-radiosChoice0"
      name="test-radios"
      onChange={[Function]}
      type="radio"
      value="test one"
    />
  </div>
  <div
    key="radio-1"
  >
    <label
      className="ter-radio__label radio-1"
      htmlFor="test-radiosChoice1"
    >
      test two
    </label>
    <input
      checked={false}
      className="radio-button-1"
      id="test-radiosChoice1"
      name="test-radios"
      onChange={[Function]}
      type="radio"
      value="test two"
    />
  </div>
  <div
    key="radio-2"
  >
    <label
      className="ter-radio__label radio-2"
      htmlFor="test-radiosChoice2"
    >
      test three
    </label>
    <input
      checked={false}
      className="radio-button-2"
      id="test-radiosChoice2"
      name="test-radios"
      onChange={[Function]}
      type="radio"
      value="test three"
    />
  </div>
</div>
`;
